### PR TITLE
Add custom cluster mgmt url flag

### PR DESCRIPTION
**Pull request by dimakis**
This allows the user to specify a custom api-gateway for their clustermgmt (ocm) which allows the CLI to install stage Addons if the targeted KFM deployment is stage.

Closes # 

### Verification Steps

1. Do x
2. Do y

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation change
- [ ] Other (please specify)

### CODE REVIEW

File: pkg/cmd/dedicated/register/registercluster.go [hunk 1]
```diff
@@ -465,11 +473,15 @@ func registerClusterWithKasFleetManager(opts *options) error {
...
+	con, err := opts.f.Config.Load()
+	if err != nil {
+		return err
+	}
+	err = createAddonWithParams(opts, getStrimziAddonIdByEnv(con), nil)
 	if err != nil {
 		return err
 	}
-	err = createAddonWithParams(opts, fleetshardAddonId, response.FleetshardParameters)
+	err = createAddonWithParams(opts, getKafkaFleetManagerAddonIdByEnv(con), response.FleetshardParameters)
```
Comment (machi1990) [inline pkg/cmd/dedicated/register/registercluster.go:484]:
```suggestion
	err = createAddonWithParams(opts, getKafkaFleetShardAddonIdByEnv(con), response.FleetshardParameters)
```

File: pkg/cmd/dedicated/register/registercluster.go [hunk 2]
```diff
@@ -439,6 +433,20 @@ func createAddonWithParams(opts *options, addonId string, params *[]kafkamgmtcli
...
 }
 
+func getStrimziAddonIdByEnv(con *config.Config) string {
+	if con.APIUrl == build.ProductionAPIURL {
+		return strimziAddonId
+	}
+	return strimziAddonIdQE
+}
+
+func getKafkaFleetManagerAddonIdByEnv(con *config.Config) string {
```
Comment (machi1990) [inline pkg/cmd/dedicated/register/registercluster.go:443]:
```suggestion
func getKafkaFleetShardAddonIdByEnv(con *config.Config) string {
```

**Review by machi1990 — COMMENTED** [review]
left one suggestion on function name.

reviewed the change, it responds to MVP requirement.
In the near term future, let's progress the conversation about KFM having a metadata API that drives most of the things here.

**Review by machi1990 — CHANGES_REQUESTED** [review]
hi @dimakis this is semi unrelated but can we sort out the wording in https://github.com/redhat-developer/app-services-cli/blob/5f68f800ab0d67d90c34301aede29fc9a767e838/pkg/core/localize/locales/en/cmd/dedicated.en.toml#L29

it should read `Kafkas` and not `Kakfas`  thanks ;-)

**Review by machi1990 — CHANGES_REQUESTED** [review]
The lines https://github.com/redhat-developer/app-services-cli/blob/5f68f800ab0d67d90c34301aede29fc9a767e838/pkg/cmd/dedicated/register/registercluster.go#L386-L390 are inversed.

It should be 
```go
if accessKafkasViaPublicNetwork {
		opts.accessKafkasViaPrivateNetwork = true
	} else {
		opts.accessKafkasViaPrivateNetwork = false
	}
```
or simply 
```go
opts.accessKafkasViaPrivateNetwork=accessKafkasViaPublicNetwork
```

While we are at it, the whole method can be refactored to
```go
prompt := &survey.Confirm{
		Message: opts.f.Localizer.MustLocalize("dedicated.registerCluster.prompt.selectPublicNetworkAccess.message"),
		Help:    opts.f.Localizer.MustLocalize("dedicated.registerCluster.prompt.selectPublicNetworkAccess.help"),
		Default: opts.accessKafkasViaPrivateNetwork,
	}
	err := survey.AskOne(prompt, &opts.accessKafkasViaPrivateNetwork)
	if err != nil {
		return err
	}
return nil
```

**Comment by dimakis** [comment]
> 

do you suspect that a private network will be the default? the reason I have that as is, is that I presumed public would be default. This was also mentioned by Keith Wall. I'm easy either way really.

**Comment by machi1990** [comment]
> > 
> 
> @machi1990 do you suspect that a private network will be the default? the reason I have that as is, is that I presumed public would be default. This was also mentioned by Keith Wall. I'm easy either way really.

Hi @dimakis there is a bug in that logic which I highlighted. If someone choose to allow for private kafkas - i.e `accessKafkasViaPublicNetwork set to "true"` then with the logic current in place, it'll send the value as `false`. And vice versa.

**Comment by machi1990** [comment]
> > 
> 
> @machi1990 do you suspect that a private network will be the default? the reason I have that as is, is that I presumed public would be default. This was also mentioned by Keith Wall. I'm easy either way really.

Also note: the suggested fix does not change the default value.

**Comment by dimakis** [comment]
> > @machi1990 do you suspect that a private network will be the default? the reason I have that as is, is that I presumed public would be default. This was also mentioned by Keith Wall. I'm easy either way really.
> 
> Hi @dimakis there is a bug in that logic which I highlighted. If someone choose to allow for private kafkas - i.e `accessKafkasViaPublicNetwork set to "true"` then with the logic current in place, it'll send the value as `false`. And vice versa.

@machi1990 It is not a bug, its a feature 😛 
The wording is changed to public, i.e. `would you like your kafkas to have access to a public network` hence the inversion.

**Comment by machi1990** [comment]
> > > @machi1990 do you suspect that a private network will be the default? the reason I have that as is, is that I presumed public would be default. This was also mentioned by Keith Wall. I'm easy either way really.
> > 
> > 
> > Hi @dimakis there is a bug in that logic which I highlighted. If someone choose to allow for private kafkas - i.e `accessKafkasViaPublicNetwork set to "true"` then with the logic current in place, it'll send the value as `false`. And vice versa.
> 
> @machi1990 It is not a bug, its a feature stuck_out_tongue The wording is changed to public, i.e. `would you like your kafkas to have access to a public network` hence the inversion.
> 
> it is a little confusing yes, as the var in the endpoint is `private`

Right. That's confusing. And the fact that it defaults to `no` public access is also confusing.

File: pkg/cmd/dedicated/register/registercluster.go [hunk 3]
```diff
@@ -384,17 +378,14 @@ func selectAccessPrivateNetworkInteractivePrompt(opts *options) error {
...
-	accessKafkasViaPublicNetwork := false
-	err := survey.AskOne(prompt, &accessKafkasViaPublicNetwork)
+	accessFromPublicNetwork := true
+	err := survey.AskOne(prompt, accessFromPublicNetwork)
 	if err != nil {
 		return err
 	}
-	if accessKafkasViaPublicNetwork {
+	if !accessFromPublicNetwork {
 		opts.accessKafkasViaPrivateNetwork = false
```
Comment (machi1990) [inline pkg/cmd/dedicated/register/registercluster.go:387]:
```suggestion
opts.accessKafkasViaPrivateNetwork = !accessFromPublicNetwork
```

File: pkg/cmd/dedicated/register/registercluster.go [hunk 4]
```diff
@@ -384,17 +378,13 @@ func selectAccessPrivateNetworkInteractivePrompt(opts *options) error {
...
+	err := survey.AskOne(prompt, accessFromPublicNetwork)
 	if err != nil {
 		return err
 	}
-	if accessKafkasViaPublicNetwork {
-		opts.accessKafkasViaPrivateNetwork = false
-	} else {
-		opts.accessKafkasViaPrivateNetwork = true
+opts.accessKafkasViaPrivateNetwork = !accessFromPublicNetwork
 	}
```
Comment (machi1990) [inline pkg/cmd/dedicated/register/registercluster.go:387]:
```suggestion
```

Comment (machi1990) [reply pkg/cmd/dedicated/register/registercluster.go:387]: trailing brace

File: pkg/core/localize/locales/en/cmd/dedicated.en.toml [hunk 5]
```diff
@@ -26,7 +26,7 @@ one = 'The ID of the OpenShift cluster to register:'
 one = 'Select the ready cluster to register'
 
 [dedicated.registerCluster.prompt.selectPublicNetworkAccess.message]
-one = 'Would you like your Kakfas to be accessible via a public network?'
+one = 'Would you like your Kafkas to be accessible via a public network?'
```
Comment (rkpattnaik780) [inline pkg/core/localize/locales/en/cmd/dedicated.en.toml:29]: Wondering if we should use `Kafka instances` instead of `Kafkas`

**Comment by jackdelahunt** [comment]
Closed because work is done in other pr #1794

**Pull request closed without merging**